### DISCUSSION TITLE
How to pass jvmargs to the build server for gradle

### DISCUSSION THREAD

**Discussion by timreebow** [microsoft/build-server-for-gradle#177] [Q&A]
Hi, 
I am a have problems using the build server for gradle. I work on a multi project gradle project, where we have a plugin which validates jvmargs from gradle.properties. This fails when I enable the bsg in vscode, since the gradle.properties file from the root directory seems not be to considered when using bsg. 

I saw this in the readme: 
> A [Preferences](https://github.com/microsoft/build-server-for-gradle/blob/develop/server/src/main/java/com/microsoft/java/bs/core/internal/model/Preferences.java) object can be put into the data field of the build/initialize request for customization. Please check the comments in the code for the meaning of each preference.
https://github.com/microsoft/build-server-for-gradle?tab=readme-ov-file#preferences

But I don't know how you set something in build/initialize. Could you explain how this could be done? Or why are the properties not being considered in bsg?

Thank you! :)

**Comment by jdneo** [comment]
Hi @timreebow

Here is an example to use `Preferences` for customization: https://github.com/microsoft/vscode-gradle/blob/895a1fee779cdde3428fa5ac63886ac0de3d5101/extension/jdtls.ext/com.microsoft.gradle.bs.importer/src/com/microsoft/gradle/bs/importer/GradleBuildServerProjectImporter.java#L173C39-L175

BTW, I'm surprised that `gradle.properties` is not considered. What's kind of properties are you specified?

**Comment by timreebow** [comment]
Hm sorry I am not an gradle expert nor gbs, still don't know how I would tell the gradle build server in vscode to have custom properties. 
thats the log from the gradle build server:

> [INFO - 18:03:52] Use build kind: WRAPPER
> [INFO - 18:03:52] Gradle version: 8.6
> [INFO - 18:03:52] Sending response 'build/initialize'. Processing request took 22 ms.
> The supplied build action failed with an exception.
> 
> FAILURE: Build failed with an exception.
> 
> * Where:
> Script 'xxx' line: 2
> 
> * What went wrong:
> A problem occurred evaluating script.
> > Failed to apply plugin 'xxx(customPlugin)'.
>    > Minimum JVM arguments are not satisfied (-Xmx1024m

So for me it seems like the gradle.properties are not picked up correctly, since when I open this project with intellij, everything works fine.

**Reply by jdneo** [reply]
I'll do some investigation, using `org.gradle.jvmargs=-Xmx1024m`. Please stay in tune.

**Reply by jdneo** [reply]
Hi, please take a look at https://github.com/microsoft/build-server-for-gradle/pull/178#issuecomment-2262412894. We might need more information to reproduce this issue.
